service-base/src/main/java/org/eclipse/hono/service/http/AbstractHttpEndpoint.java

```diff
@@ -347,6 +349,37 @@ protected final String getDeviceIdParam(final RoutingContext ctx) {
         return ctx.request().getParam(PARAM_DEVICE_ID);
     }
 
+    /**
+     * Get request parameter value and check if it has been set. If it's not set, fail the request.
+     *
+     * @param paramName The name of the parameter to get.
+     * @param ctx The routing context of the request.
+     * @return The value of the parameter if it's set or {@code null} otherwise.
+     * @throws NullPointerException If ctx is {@code null}.
+     */
+    protected final String getRequestParam(final String paramName, final RoutingContext ctx) {
```

Comment:
maybe this should then be named *getMandatoryRequestParameter* in order to more easily distinguish from the variant accepting the boolean param?